src/python/pants/backend/typescript/target_types.py

```diff
@@ -26,7 +26,7 @@ class TypeScriptDependenciesField(Dependencies):
     pass
 
 
-class TypeScriptSourceField(SingleSourceField):
+class TypeScriptSourceField(JSRuntimeSourceField):
```

Comment:
I wonder if this change has broader consequences, where other fieldsets that are requesting JS-like sources will implicitly start picking up TS ones too? 

This may be desirable (e.g. many of the other child issues of #22147 are "make TS behave like JS", essentially), but presumably we should be intentional about the best way to go about this.

I have a vague memory that @tobni may have done some thinking about this previously?